src/systems/filecoin_markets/storage_market/storage_market_actor_state.go

```diff
@@ -253,19 +296,16 @@ func (st *StorageMarketActorState_I) _getStorageFeeSinceLastPayment(rt Runtime,
 
 }
 
-func (st *StorageMarketActorState_I) _slashDealCollateral(rt Runtime, dealP deal.StorageDealProposal) actor.TokenAmount {
-	amountToSlash := dealP.ProviderBalanceRequirement()
-	providerBal := st._safeGetBalance(rt, dealP.Provider())
+func (st *StorageMarketActorState_I) _rtSlashDealCollateral(rt Runtime, dealP deal.StorageDealProposal) actor.TokenAmount {
+	Assert(st._addressEntryExists(dealP.Provider()))
 
-	if providerBal.Locked() < amountToSlash {
-		amountToSlash = providerBal.Locked()
-		// TODO: decide on error handling here
-		panic("TODO")
-	}
+	slashAmount := dealP.ProviderBalanceRequirement()
```

Comment:
This may slash more than is required. It doesn't account for the portion of the deal that is already past. If that is intentional comment plz :)

Reply:
Thanks for flagging, I think the intended behavior is to slash all deal collateral when a sector is terminated. We might potentially change this to a function of the portion of the deal that is already past but that should be a fairly simple change.